Sources/ArcGISToolkit/Components/Compass/Compass.swift

```diff
@@ -110,10 +114,37 @@ public struct Compass: View {
                                  """
                     )
                 )
+                .snapToZeroSensoryFeedback(enabled: snapToZeroSensoryFeedbackEnabled, heading: heading)
         }
     }
 }
 
+private extension View {
+    /// Enables the snap to zero sensory feedback
+    /// when it is available.
+    @ViewBuilder
+    func snapToZeroSensoryFeedback(enabled: Bool, heading: Double) -> some View {
+        if #available(iOS 17.0, *) {
+            if enabled {
+                sensoryFeedback(.selection, trigger: heading) { oldValue, newValue in
+                    if (!oldValue.isZero && newValue.isZero) ||
+                        (oldValue.isZero && !newValue.isZero) {
```

Comment:
I'm on the fence about snapping when coming off zero. 

I can't quite put my finger on it, but I think the slight delay in the feedback once you're already off zero is perceivable.

Do you have any thoughts on having it just when entering zero?

Reply:
The perceptible delay seems to me to be in when the compass starts animating to visible from coming off zero. The haptic seems to coincide well with when the map rotates, IMO.

I tested it both ways and both seem good to me. I could go either way.

The [HIG](https://developer.apple.com/design/human-interface-guidelines/playing-haptics#Selection) doesn't give a lot of guidance:
> Selection haptics provide feedback while the values of a UI element are changing.